testing/matrix-sdk-test/src/event_factory.rs

```diff
@@ -1003,15 +1003,21 @@ impl EventFactory {
         self.event(CallInviteEventContent::new(call_id, lifetime, offer, version))
     }
 
-    /// Create a new `m.call.notify` event.
-    pub fn call_notify(
+    /// Create a new `m.rtc.notification` event.
+    pub fn rtc_notification(
         &self,
-        call_id: String,
-        application: ApplicationType,
-        notify_type: NotifyType,
-        mentions: Mentions,
-    ) -> EventBuilder<CallNotifyEventContent> {
-        self.event(CallNotifyEventContent::new(call_id, application, notify_type, mentions))
+        notification_type: NotificationType,
+        mentions: Option<Mentions>,
+        rtc_membership_event_id: Option<OwnedEventId>,
+    ) -> EventBuilder<RtcNotificationEventContent> {
+        let mut content = RtcNotificationEventContent::new(
+            MilliSecondsSinceUnixEpoch::now(),
+            Duration::new(20, 0),
+            notification_type,
+        );
+        content.relates_to = rtc_membership_event_id.map(Reference::new);
+        content.mentions = mentions;
```

Comment:
The more idiomatic way with the `EventFactory` builder pattern would be to add `fn mentions()` and `fn rtc_membership_event()` methods on `impl EventBuilder<RtcNotificationEventContent>`, so that we don't have `Option`al parameters in this ctor. Also, I would rather make it so that the timestamp and durations are configurable (aka function parameters), so we don't have tight coupling between the test framework and the tests themselves (or, at least let's rename `rtc_notification` to `predefined_rtc_notification` or something like that)

Reply:
Thanks for all the tips! I missed the fact that one could do `impl EventBuilder<CONTENT>`. 
Done here https://github.com/matrix-org/matrix-rust-sdk/pull/5668/commits/0505ce28c3fbf4a11e93a61f913b178eb23d9412